store/db_test.go

```diff
@@ -78,6 +78,9 @@ func createTable(db *DB, t *testing.T) {
 }
 
 func TestDBRace(t *testing.T) {
+	// TODO(sgotti) this will not find concurrenct accesses to ql db from
+	// multiple processes using multiple goroutines. A test that spawns at
```

Comment:
@jonboulle Sorry, it's difficult to explain. What I meant was that:

Given that the store db model is based on making short lived Open/Tx/Close to make the ql db accessible by multiple processes and/or goroutines without long blocking and possible deadlocks.

The "cannot acquire lock: resource temporarily unavailable" doesn't happen with:

* Multiple OS processes **not** having multiple goroutines using a shared store instance.
* Only one OS process accessing the db (also if it has multiple goroutines using a shared store instance like this test and the rkt api-service).

It happens only when there are multiple OS processes concurrently accessing the ql db and at least one of them has multiple goroutines using a shared store instance.


Plus, after more analysis, I think that the case
* Only one OS process accessing the db (also if it has multiple goroutines using a shared store instance like this test and the rkt api-service).
that in the commit comments I say won't give an error but cause possible db corruption is probably not true so I'll remove it from the commit comment.